lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -130,18 +126,26 @@ define((require, exports, module) ->
         @lastChangeTime = null
       else
         @currentDelayedUpdateId = null
-        if @contextPreviewExists
-          { row: cursorRow, column: cursorColumn } = @editor.getCursorPosition()
-          curContext = LatexParsingContext.getContext(@editor.getSession(), cursorRow, cursorColumn)
-          if curContext != "equation"
-            @destroyContextPreview()
-          else
-            @updateRange()
+        { row: cursorRow, column: cursorColumn } = @editor.getCursorPosition()
+        curContext = LatexParsingContext.getContext(@editor.getSession(), cursorRow, cursorColumn)
+
+        if curContext == "equation"
+          @updateRange()
+        else
+          @destroyRange()
```

Comment:
btw, calls of `destroyRange` and `disableUpdates` also correlate. Any sense to replace with a single method and call? I am not absolutely sure because you may keep in mind scenario where they are called independently or maybe just want to see instructions explicitly even if they correlate